CPT: Display post author in custom post types list

aduth opened the pull request:
This pull request seeks to add the post's author to the meta information on the [custom post types list screen](https://wpcalypso.wordpress.com/types/post). ~~It also enables changing the author inline, making use of the existing [`<AuthorSelector />` component](https://github.com/Automattic/wp-calypso/tree/master/client/components/author-selector) (used also in the post editor).~~ (Removed per feedback)

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/16742724/ee2d6510-4776-11e6-85f9-f9f37fdfe3d9.png)|![After](https://cloud.githubusercontent.com/assets/1779930/16766699/8aaf16ee-4809-11e6-81b2-ea5ec784dfa9.png)

[_See original concept_](https://cloud.githubusercontent.com/assets/1779930/16742674/b9d91c3c-4776-11e6-870f-6efa76614e65.png)

[_Original version with author selector_](https://cloud.githubusercontent.com/assets/1779930/16742730/f6057912-4776-11e6-9c62-773f09cd0a48.png)

__Testing instructions:__

Verify that you are able to view the author of a post, and reassign the author if you have capabilities for doing so.

1. Navigate to the [custom post types list screen](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Select a filter where posts exist
4. Note that the author is shown~~, with Gravatar,~~ under the post title
5. ~~If capabilities exist, note that a chevron is shown adjacent the author to select a new author~~
6. ~~Select a new author~~
7. ~~Note that after a brief delay, the author is changed and persists through page refresh~~

__Caveats / Follow-up Tasks:__

_These caveats are no longer applicable after removal of author selector per feedback._

- The change in author is not reflected instantaneously, because the format of `author` is different between reading and updating endpoints. This should be resolved by normalizing the post object's author in the `getNormalizedPost` selector.
- The author selector does not account for the capabilities of the author to which the post will be assigned. Even if a user does not have capabilities of editing a particular post type, it will still be presented as an option in the dropdown. This is an issue of the underlying `<AuthorSelector />` component.

Test live: https://calypso.live/?branch=add/cpt-assign-author

folletto commented:
I'm not sure if we should make the author changeable from here, especially since it won't have a confirmation. 

That said, if we really think we want to have the ability to change it in there, I feel that visually should match the original designs:

![screen shot 2016-07-11 at 20 07 55](https://cloud.githubusercontent.com/assets/4389/16743223/34825cb0-47a3-11e6-8225-dcb08d8cce6e.png)

hugobaeta commented:
@folletto and I were just discussing this, I agree with his comment: I feel unnecessary to provide that functionality in the card - this is meant to be a quick overview, before the user takes the primary action of editing the entry.

The avatar, even tho it's a logical addition, feels too heavy in that meta information list - specially in the second position, it adds a visual weight to the middle of the card that takes attention away from the intended hierarchy in the card). Also, at such small size, it becomes a blur :) Let's ditch it and use the icon I used in the mockup, please.

nylen commented:
I also think providing the ability to change authorship here is more confusing than helpful.  Changing the author of a post is a very uncommon operation (0.08% of `editor_actions` stat).

aduth commented:
Thanks for the feedback. I removed the option to reassign the author and the author's Gravatar.

Updated screenshot:

![image](https://cloud.githubusercontent.com/assets/1779930/16766699/8aaf16ee-4809-11e6-81b2-ea5ec784dfa9.png)

folletto commented:
Checked on Chrome, design wise it's 👍

aduth commented:
>(We should remove the "(last updated)" bit, but can be done in a separate PR, no need here)

@hugobaeta mentioned this once before, and it's always been part of the underlying `<PostRelativeTimeStatus />` component for draft posts ([reference](https://github.com/Automattic/wp-calypso/blob/ab93c61b8003801763b7d4e987b2b322a8d30d75/client/my-sites/post-relative-time-status/index.jsx#L39)), originally from the posts screen. Since we effectively did away with a dedicated post drafts screen, I think we could remove this without it affecting anything else?

folletto commented:
I think we can get away from that, we might replace later with a tooltip, but the date is the latest update (or, publish in case of published posts) so it should be contexually clear what that is. :)

aduth commented:
See #6707 for removal of last modified label.

nylen commented:
Looks good & works well.  Could use a rebase but otherwise good to go.
